Ka-Block.safariextension/start.js

```diff
@@ -1,32 +1,36 @@
-// Remember domain filter results.
-const cache = new Map()
+// Content Blocker polyfill for Safari 8 and under
+if (!('onwebkitmouseforcedown' in window)) {
+  // Remember domain filter results.
+  const cache = new Map()
 
-// Check with global page for approval.
-function blocked(event, url) {
-  var result = cache.get(url.hostname)
-  if (result === undefined) {
-    result = safari.self.tab.canLoad(event, url.href) === 'block'
-    cache.set(url.hostname, result)
+  // Check with global page for approval.
+  function blocked(event, url) {
+    var result = cache.get(url.hostname)
+    if (result === undefined) {
+      result = safari.self.tab.canLoad(event, url.href) === 'block'
+      cache.set(url.hostname, result)
+    }
+    return result
   }
-  return result
-}
 
-// Filter all external resource requests.
-document.addEventListener('beforeload', function(event) {
-  // Resolve relative URLs.
-  const url = new URL(event.url, window.location.href)
+  // Filter all external resource requests.
+  document.addEventListener('beforeload', function(event) {
+    // Resolve relative URLs.
+    const url = new URL(event.url, window.location.href)
 
-  // Skip about:blank, data:image, etc.
-  if (!url.origin || url.origin === 'null') {
-    return
-  }
+    // Skip about:blank, data:image, etc.
+    if (!url.origin || url.origin === 'null') {
+      return
+    }
 
-  // Allow same origin resources.
-  if (window.location.origin === url.origin) {
-    return
-  }
+    // Allow same origin resources.
+    if (window.location.origin === url.origin) {
+      return
+    }
 
-  if (blocked(event, url)) {
-    event.preventDefault()
-  }
-}, true)
+    if (blocked(event, url)) {
+      console.info('Content blocker prevented frame displaying ' + window.location.href + ' from loading a resource from ' + event.url)
```

Comment:
Native content blockers log a message similar to this in the console. I choose to emulate it as close as possible for debugging.